From a1e01aeb50807fc16ca988663f8ea713f182c4d7 Mon Sep 17 00:00:00 2001
From: Maciej Trela <maciej.trela@intel.com>
Date: Mon, 12 Mar 2012 23:29:30 +0000
Subject: [PATCH 30/30] isci: enable BCN in sci_port_add_phy()

7191bb5a3c52917b653ddd27d29b637961cd5930 of isci.git
git://git.kernel.org/pub/scm/linux/kernel/git/djbw/isci.git

Ensure we enable receiving BCN's from the
hardware when adding phy to isci_port.
Otherwise if we get BCN before the port is
created we won't see any BCN

Signed-off-by: Maciej Trela <maciej.trela@intel.com>
Reported-by: Richard Boyd <richard.g.boyd@intel.com>
Signed-off-by: Dan Williams <dan.j.williams@intel.com>
---
 drivers/scsi/isci/port.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/scsi/isci/port.c b/drivers/scsi/isci/port.c
index 6ef4bd9..0a3aec1 100644
--- a/drivers/scsi/isci/port.c
+++ b/drivers/scsi/isci/port.c
@@ -1199,6 +1199,8 @@ enum sci_status sci_port_add_phy(struct isci_port *iport,
 	enum sci_status status;
 	enum sci_port_states state;
 
+	sci_port_bcn_enable(iport);
+
 	state = iport->sm.current_state_id;
 	switch (state) {
 	case SCI_PORT_STOPPED: {
-- 
1.7.9.7

